### DISCUSSION TITLE
in v6 with global state

### DISCUSSION THREAD

**Discussion by ravenkim** [remix-run/react-router#11604] [Proposals]
I want to put location data into redux state. But i cannot use like this 
![image](https://github.com/remix-run/react-router/assets/41793830/26718ca0-b212-4e3a-891a-73f3fd17890e)

because useLocation must be inside of RouterProvider

so i suggest to put children in <RouterProvider/>  to use like this 
![image](https://github.com/remix-run/react-router/assets/41793830/b4645923-d0e3-43b3-a840-91ab4373f225)

![image](https://github.com/remix-run/react-router/assets/41793830/2f383a42-90af-4667-974d-7497a7f92a66)

**Comment by sergiodxa** [comment]
Why do you need it in Redux? You could call useLocation in any component you nee dit, it's already a global state.

Aside of that, adding it inside your router and using an effect to send them to Redux makes sense, you need to keep updating the location on your Redux store whenever the actual location from useLocation changes, it's the same as if you stored it in a useState, you would have to do:

```ts
let location = useLocation()
let [state, setState] = useState(() => location)
useEffect(() => {
  setState(location)
}, [location])
```

Which leads me to my first question, you could just use the useLocation directly instead of saving it in another place.

**Reply by ravenkim** [reply]
thanks for answer, 
I know it is already a global state. But for commonality I want to handle every global state in redux devtools
![image](https://github.com/remix-run/react-router/assets/41793830/822b83bb-2a28-40f5-b4ac-cca62e1f602e)



Now I am using globalState like this 
![image](https://github.com/remix-run/react-router/assets/41793830/20a463fd-9f9e-4e8d-acad-b71dde3ddf38)

beacuse it is easy to find out what is global, and i can see what changed in redux devtools

but if I use useLocation() then i need to put this code in every my component page. 
That's why I want to connect react-route with redux.

Then if i do not save, I'm looking for a better way to connect 'Redux' and 'Router'

Thanks for answer
